docs/quickstart/basics.rst

```diff
@@ -60,38 +78,49 @@ Request Body Validation
 
 .. code-block:: python
 
-   from flask_rebar import RequestSchema
```

Comment:
(FWIW, I haven't yet hit a case where I've wanted Rebar's `RequestSchema`; in all my use cases so far, I've preferred ["being liberal in what I accept and conservative in what I send"](https://en.wikipedia.org/wiki/Robustness_principle), at least as far as unrecognized fields are concerned.)

Reply:
I put it everywhere since I prefer a break fast strategy with a REST API. For example, if for some reason you have a breaking change in the API, you don't want to have clients sending requests thinking that all went well and it applied some changes but in reality it didn't (and no alert since its a 200).

Reply:
(To be clear, I'm by no means saying it's always wrong to fail hard for unrecognized fields, just that it's definitely not always right. Maybe more citation needed, so just did a quick search and immediately turned up something interesting: https://github.com/zalando/restful-api-guidelines/pull/248/files#diff-bf72c91a3bee08729395c486d36a5cc2R54-R58 – this topic is deep!)

Reply:
I really like this website, thanks for sharing. I think we should try to encourage some of it's principles while allowing divergences because they are quite strict and hard to meet for a lot of people.

Reply:
So looking with fresh eyes today... That API Guidelines link I posted last night is saying that the Robustness Principle applies on the client side, but on the server side, it doesn't!:
> Unknown input fields in payload or URL should not be ignored; servers should provide error feedback to clients via an HTTP 400 response code.

This is not what I thought was best practice, so I'll definitely be doing more research on this! If this API Guidelines doc is correct, then it's quite a validation of `flask_rebar`'s `RequestSchema`, and I'll correct my understanding, my code, and my teaching accordingly!

In the meantime, the current revision of this PR currently is still removing the use of `RequestSchema` for the example `request_body_schema`s. That means going from conforming to these guidelines to not conforming to them. I really don't want to be responsible for, in the process of trying to improve the docs' demonstration of best practices, actually making them _worse_ in this particular way.

At the same time, I'm very keen to see #161 happen, because I do think that having two different schemas (request and response) for the same type of object is not always necessary, and when it isn't, as in this Todo example, it adds unnecessary complexity to our docs. Shall we wait for #161 to be resolved so we can incorporate the outcome into these changes before we merge them?

Thank you @sytten for prompting this!

Reply:
We can wait, I will try to prioritize this weekend of monday. Though I am not sure if we want to only do that in V2 or backport it to V1.